### PR TITLE
feat: Use XDG Base Directory Specificationfor TESTGROUND_HOME directory

### PR DESCRIPTION
**Pull request by dougEfresh**
Having TESTGROUND_HOME directory in `~/testground` pollutes my $HOME. There are standards for application data/cache/config across operating systems, see https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html and https://github.com/adrg/xdg

For linux/unix:
`~/.config/testground`

For Mac/Darwin:

`~/Library/Application Support/testground`

### CODE REVIEW

**Comment by laurentsenta** [comment]
Thanks for the contribution @dougEfresh 
Before merging, we'll have to verify the impact on other setups (like EKS infra).

Could you add a line to the CHANGELOG.md?

File: pkg/config/loader.go
Comment (laurentsenta) [inline pkg/config/loader.go:77]: I think you want to leave read and execute here, could you clarify why we want a `0`?

Comment (dougEfresh) [reply pkg/config/loader.go:77]: I personally do not like world readable directories, but I'll change it back.

**Comment by dougEfresh** [comment]
Could someone test this on a mac? I only run linux :)

**Comment by dougEfresh** [comment]
> * Could you also update related documentation? like: https://docs.testground.ai/v/master/table-of-contents/getting-started#running-testground

See https://github.com/testground/docs/pull/24

File: CHANGELOG.md
Comment (laurentsenta) [inline CHANGELOG.md:15]:
```suggestion
- Move default TESTGROUND_HOME from `~/testground` to xdg directory specification. See [PR 1544]
```

File: install.sh
Comment (laurentsenta) [inline install.sh:23]: @galargh would you have time to look into this part? It's supposed to run in CI as well.

File: CHANGELOG.md
Comment (dougEfresh) [reply CHANGELOG.md]: Also add link to bottom of CHANGELOG

**Comment by galargh** [comment]
Hey! Thanks for proposing the change 😄 

Before we proceed with merging it straight away, I'd want us to consider the impact it might have on the current testground users. From what I gather, if someone is using testground now, doesn't have `TESTGROUND_HOME` set and upgrades to the version with this change in it, then they're imported plans won't work anymore. As such, in my opinion it has to be treated and communicated as a breaking change. And if so, do we want to introduce it on its' own or would we rather batch it up with other breaking changes?

**Comment by dougEfresh** [comment]
> As such, in my opinion it has to be treated and communicated as a breaking change. And if so, do we want to introduce it on its' own or would we rather batch it up with other breaking changes?

How about we make this backward compatible ?  if `~/testground` exists, just use that. Shouldn't be so hard to implement.

**Comment by dougEfresh** [comment]
@galargh I updated this PR to check for legacy directory $HOME/testground and use that.

**Review by galargh — COMMENTED** [review]
Thank you, Douglas, for the changes! The check for `~/testground` existence addresses my initial concerns well :) 

I added the last set of small review comments. I think we'll be ready to merge after the next iteration 🥳

File: pkg/config/loader.go
Comment (dougEfresh) [reply pkg/config/loader.go:70]: I thought about this  as well. I don't think at this point in the code to check a specific error. If there is some issue, then later on in the code  `switch fi, err := os.Stat(home);` will catch it. 
But I will add is a check that stat us a Directory.

File: pkg/config/loader.go
Comment (dougEfresh) [reply pkg/config/loader.go:68]: Added a function to check legacy dir or use default home. I think this makes the code more clear.  Plus, easy to remove when the time comes to deprecate $HOME/testground 
Let me know what you think.

Comment (dougEfresh) [reply pkg/config/loader.go:68]: Also, one note of interest, `os.UserHomeDir()` has been replaces with `xdg.Home` . Which defaults to `/` if `$HOME` not set.  I think this is acceptable, but wanted to point this out to you.

**Comment by galargh** [comment]
I moved the call to the function you created so that we don't print the warning if the default is not used. I also fixed the test setup and made the warning more explicit. 

All looks good now. I'll test it on mac just to be sure and merge. Thank you for your contribution :)

**Review by galargh — APPROVED** [review]

**Pull request merged**
